### PR TITLE
Add removeEventCallback API

### PR DESCRIPTION
**Pull request by tnorling**
- Adds an API to remove an event callback
- Changes `eventCallbacks` array to a Map with a guid key
- `addEventCallback` returns an id
- Fixes login vs acquireToken logic which had the comparison backwards

### CODE REVIEW

File: lib/msal-browser/src/app/ClientApplication.ts [hunk 1]
```diff
@@ -901,11 +901,20 @@ export abstract class ClientApplication {
      * Adds event callbacks to array
      * @param callback 
      */
-    addEventCallback(callback: EventCallbackFunction) {
+    addEventCallback(callback: EventCallbackFunction): string | null {
         if (this.isBrowserEnvironment) {
             this.logger.verbose("Event callback registered");
```
Comment (jasonnutter) [inline lib/msal-browser/src/app/ClientApplication.ts:906]: Can this log message included the `callbackId`?

File: lib/msal-browser/src/app/ClientApplication.ts [hunk 2]
```diff
@@ -901,11 +901,20 @@ export abstract class ClientApplication {
...
+            this.eventCallbacks.set(callbackId, callback);
+
+            return callbackId;
         }
+
+        return null;
+    }
+
+    removeEventCallback(callbackId: string): void {
+        this.eventCallbacks.delete(callbackId);
```
Comment (jasonnutter) [inline lib/msal-browser/src/app/ClientApplication.ts:917]: Log message here as well.

**Review by jo-arroyo — APPROVED** [review]
LGTM!

File: lib/msal-browser/src/app/ClientApplication.ts (hunk 1 above)
Comment (tnorling) [reply lib/msal-browser/src/app/ClientApplication.ts]: Done

File: lib/msal-browser/src/app/ClientApplication.ts (hunk 2 above)
Comment (tnorling) [reply lib/msal-browser/src/app/ClientApplication.ts:917]: Done

**Review by jasonnutter — APPROVED** [review]

**Review by pkanher617 — APPROVED** [review]

**Pull request merged**